### PR TITLE
Feature / Rewards Logic architecture

### PR DESCRIPTION
**Pull request by scorpioborn**
## Description

---

## Type of change

Please check the options that are relevant. This PR introduces

- [ ] Patch: Bug fix (non-breaking change which fixes an existing issue)
- [ ] Minor: New feature (non-breaking change which adds new functionality)
- [ ] Major: Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] This change requires a documentation update

## Features Description

- [ ] NA

---

## Test Cases

- [ ] NA

---

## Documentation

---

## Checklist:

- [ ] I have added change type (major|minor|patch) in the PR
- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] I have added tests that prove my fix is effective or that my feature works
- [ ] New and existing unit tests pass locally with my changes
---

### CODE REVIEW

File: proto/sge/reward/campaign.proto [hunk 1]
```diff
@@ -18,45 +18,52 @@ message Campaign {
     json_name = "uid"
   ];
 
+  // house_uid is the unique identifier of the promoter house.
+  string house_uid = 3 [
```
Comment (3eyedraga) [inline proto/sge/reward/campaign.proto:22]: it is not required. Promoter which is added below solves the problem. House id is not required.

File: proto/sge/reward/query.proto [hunk 2]
```diff
@@ -21,7 +21,7 @@ service Query {
 
   // Queries a specific Campaign item.
   rpc Campaign(QueryCampaignRequest) returns (QueryCampaignResponse) {
-    option (google.api.http).get = "/sge-network/sge/reward/campaign/{uid}";
+    option (google.api.http).get = "/sge-network/sge/reward/campaign/{house_uid}/{uid}";
```
Comment (3eyedraga) [inline proto/sge/reward/query.proto:24]: since house id is not required this change also is not required

File: proto/sge/reward/tx.proto [hunk 3]
```diff
@@ -22,8 +22,10 @@ message MsgCreateCampaign {
   string creator = 1;
   // uid is the unique identifier of the campaign.
   string uid = 2;
+  // house_uid is the unique identifier of the promoter house.
+  string house_uid = 3;
```
Comment (3eyedraga) [inline proto/sge/reward/tx.proto:26]: remove house_id

Comment (3eyedraga) [reply proto/sge/reward/tx.proto:26]: remove house_id from everywhere

Comment (scorpioborn) [reply proto/sge/reward/tx.proto]: removed, temporarily

File: proto/sge/reward/query.proto (hunk 2 above)
Comment (scorpioborn) [reply proto/sge/reward/query.proto]: removed, temporarily

File: proto/sge/reward/campaign.proto (hunk 1 above)
Comment (scorpioborn) [reply proto/sge/reward/campaign.proto]: removed, temporarily

**Review by 3eyedraga — APPROVED** [review]

**Pull request merged**